app/src/main/java/com/kickstarter/ui/fragments/projectpage/ProjectCampaignFragment.kt

```diff
@@ -44,13 +51,23 @@ class ProjectCampaignFragment :
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         super.onCreateView(inflater, container, savedInstanceState)
+
+        this.context?.getEnvironment()?.let { env ->
+            viewModelFactory = ProjectCampaignViewModel.Factory(env)
+        }
+
         binding = FragmentProjectCampaignBinding.inflate(inflater, container, false)
         return binding?.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        viewElementAdapter = ViewElementAdapter(requireActivity(), this, this.lifecycle())
+
+        viewElementAdapter = ViewElementAdapter(requireActivity(), this)
+        viewElementAdapter.let {
+            lifecycle.addObserver(it as LifecycleObserver)
```

Comment:
This line adds the ViewElementAdapter as an observer to this Fragment's lifecycle